src/Chapter3.hs

```diff
@@ -374,6 +383,42 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight = MkKnight
+    { knightHealth  :: Int
+    , knightAttack  :: Int
+    , knightGold    :: Int
+    }
+
+data Monster = MkMonster
+    { monsterHealth  :: Int
+    , monsterAttack  :: Int
+    , monsterGold    :: Int
+    }
+
+
+fight :: Knight -> Monster -> Int
+fight knight monster
+    | (knightAttack knight) >= (monsterHealth monster) = (knightGold knight) + (monsterGold monster)
```

Comment:
In Haskell, you should not use parenthesis in every expression, only where necessary for the parser to separate some cases. In here, everything can be without parenthesis 🙂 
```suggestion
    | knightAttack knight >= monsterHealth monster = knightGold knight + monsterGold monster
```